src/gf.c

```diff
@@ -1039,9 +1056,35 @@ jl_lambda_info_t *jl_get_specialization1(jl_tupletype_t *types)
     return NULL;
 }
 
-JL_DLLEXPORT void jl_compile_hint(jl_tupletype_t *types)
+// Specialize, but return NULL if there is a call ambiguity
+jl_lambda_info_t *jl_checked_specialization1(jl_tupletype_t *types)
```

Comment:
I think this should be merged with `jl_get_specialization1`. I don't think there is a use case for getting a specialization that would be a method error (due to ambiguities) if actually called.